File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int c = 0;
        for (int i = 0; i < nums.length; i++)
            for (int j = Math.min(i + k, nums.length - 1); j > i; j--, c++) {
                if (nums[i] == nums[j])
                    return true;
                if (c > 9999)
                    return false;
            }
        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    const window = new Set();

    for(var i=0;i<nums.length;i++){
        if (window.has(nums[i])){
            return true;
        }
        window.add(nums[i]);
        if(i>=k){
            window.delete(nums[i-k]);
        }
    }
    return false;
};